test/promql/tests/test_unschedulable_pods.yaml

```diff
@@ -0,0 +1,32 @@
+# This is the main input for unit testing.
+# Only this file is passed as command line argument
+evaluation_interval: 1m
+
+rule_files:
+  - ../extracted-rules.yaml
+
+tests:
+  # Interval for the samples of the series
+  - interval: 1m
+    # Series data
+    input_series:
+
+      - series: 'kube_pod_status_unschedulable{namespace="prod", pod="prod-pod-1"}'
+        values: '1 1 1 1 1 1 1 1 1 1'
+
+      - series: 'kube_pod_status_unschedulable{namespace="prod", pod="prod-pod-2"}'
+        values: '1 1 1 1 0 0 0 0 0 0'
+
+      - series: 'kube_pod_status_unschedulable{namespace="prod-tenant", pod="prod-pod"}'
+        values: '1 1 1 1 1 1 0 0 0 0'
+
+    # Unit tests for alerts
+    alert_rule_test:
+      - eval_time: 4m
+        alertname: UnschedulablePods
+        exp_alerts:
+          - exp_labels:
+              severity: warning
+            exp_annotations:
+              summary: "Unschedulable pods"
+              description: "Alert when there are issues with pod scheduling"
```

Comment:
What kind of errors? in the alert rule itself we referred to it as Pending, are there more scenarios? let's explain them.